lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -442,31 +436,26 @@ struct TypeLoweringVisitor : public FIRRTLVisitor<TypeLoweringVisitor, bool> {
   // Cache some attributes
   const AttrCache &cache;
 
-  // Keep track of public modules.
-  const llvm::DenseSet<FModuleLike> &publicModuleSet;
+  const llvm::DenseMap<FModuleLike, Convention> &conventionTable;
```

Comment:
Since this pass is mutating module annotations concurrently across multiple threads, accessing another module's attributes directly can crash mfc. In an [earlier issue](https://github.com/llvm/circt/pull/4655), @uenoku had to switch to a side-table to track visibility, for this reason.

Reply:
I thought on this pass the module op mutation was done serially and only the module bodies done in parallel.

Reply:
Ah, yes, it was depending on parallel multation resulting in the same type.  We've generally split module update form module body update to solve this, but it's fine for now.  When module type lowering is split out to it's own pass, this whole pass can become a fmodule pass instead of a circuit pass.